pkg/io/tfio/sg.go

```diff
@@ -86,18 +86,24 @@ func sg(sgName, comment string) tf.Block {
 		Labels:  []string{quote("ibm_is_security_group"), changeScoping(quote(sgName))},
 		Comment: comment,
 		Arguments: []tf.Argument{
-			{Name: "name", Value: quote("sg-" + sgName)},
+			{Name: "name", Value: changeScoping(quote("sg-" + sgName))},
 			{Name: "resource_group", Value: "local.sg_synth_resource_group_id"},
-			{Name: "vpc", Value: "local.sg_synth_vpc_id"},
+			{Name: "vpc", Value: fmt.Sprintf("local.name_%s_id", ir.ScopingComponents(sgName)[0])},
 		},
 	}
 }
 
-func sgCollection(t *ir.SGCollection) *tf.ConfigFile {
+func sgCollection(t *ir.SGCollection, vpc string) *tf.ConfigFile {
 	var resources []tf.Block //nolint:prealloc  // nontrivial to calculate, and an unlikely performance bottleneck
-	for _, sgName := range t.SortedSGNames() {
+	var sortedCollection []ir.SGName
+	if vpc == "" {
+		sortedCollection = t.SortedSGNames()
+	} else {
+		sortedCollection = t.SortedSGNamesInVPC(vpc)
+	}
+	for _, sgName := range sortedCollection {
 		comment := ""
-		rules := t.SGs[sgName].Rules
+		rules := t.SGs[ir.ScopingComponents(string(sgName))[0]][sgName].Rules
```

Comment:
can we make `ir.SGName` a type alias of string?

Reply:
will be done in `convert_to_ipblock` branch.

Reply:
I am not sure because it should implement the `String()` method

Reply:
If `ir.SGName` becomes an alias of `string`, you will no longer need the `String()` method. However other problems will arise, as `ir.SGName` is being used in a type-switch and in type assertions. An aliased type cannot be used in these. So let's leave it as-is.